sql/core/src/main/scala/org/apache/spark/sql/streaming/DataStreamWriter.scala

```diff
@@ -300,54 +301,54 @@ final class DataStreamWriter[T] private[sql](ds: Dataset[T]) {
         "write files of Hive data source directly.")
     }
 
-    if (source == "memory") {
+    if (source == SOURCE_NAME_TABLE) {
+      assertNotPartitioned("table")
+
+      import df.sparkSession.sessionState.analyzer.{NonSessionCatalogAndIdentifier, SessionCatalogAndIdentifier}
+
+      import org.apache.spark.sql.connector.catalog.CatalogV2Implicits._
+      val tableInstance = df.sparkSession.sessionState.sqlParser
+        .parseMultipartIdentifier(tableName) match {
+
+        case NonSessionCatalogAndIdentifier(catalog, ident) =>
+          catalog.asTableCatalog.loadTable(ident)
+
+        case SessionCatalogAndIdentifier(catalog, ident) =>
```

Comment:
shall we just use `CatalogAndIdentifier`?

Reply:
Is it OK to skip the namespace length check in SessionCatalogAndIdentifier here?

Reply:
It's OK. `V2SessionCatalog.loadTable` checks the namespace as well.